sys/net/gnrc/network_layer/sixlowpan/frag/minfwd/gnrc_sixlowpan_frag_minfwd.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+
+#include "net/gnrc/netif/hdr.h"
+#include "net/gnrc/pktbuf.h"
+#include "net/gnrc/sixlowpan/internal.h"
+#include "utlist.h"
+
+#include "net/gnrc/sixlowpan/frag/minfwd.h"
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+static gnrc_pktsnip_t *_netif_hdr_from_vrbe(const gnrc_sixlowpan_frag_vrb_t *vrbe)
+{
+    gnrc_pktsnip_t *res = gnrc_netif_hdr_build(NULL, 0, vrbe->super.dst,
+                                               vrbe->super.dst_len);
+    if (res == NULL) {
+        DEBUG("6lo minfwd: can't allocate netif header for forwarding.\n");
+        return NULL;
+    }
+    gnrc_netif_hdr_set_netif(res->data, vrbe->out_netif);
+    return res;
+}
+
+static inline bool _is_last_frag(const gnrc_sixlowpan_frag_vrb_t *vrbe)
+{
+    return (vrbe->super.current_size >= vrbe->super.datagram_size);
+}
+
+int gnrc_sixlowpan_frag_minfwd_forward(gnrc_pktsnip_t *pkt,
+                                       const sixlowpan_frag_n_t *frag,
+                                       gnrc_sixlowpan_frag_vrb_t *vrbe,
+                                       unsigned page)
+{
+    sixlowpan_frag_t *new;
+    gnrc_pktsnip_t *tmp;
+    const size_t fragsnip_size = sizeof(*frag) -
+                                 ((sixlowpan_frag_1_is((sixlowpan_frag_t *)frag))
+                                 ? sizeof(frag->offset) : 0U);
```

Comment:
What is `frag->offset` and why is it only stored in the first fragment?

Reply:
It is _not_ stored in the first fragment, that's why it is subtracted. I won't go into 6LoWPAN header format basics here in a comment, but I agree that this code is somewhat unclear and reversed logic. Will fix to
```suggestion
    const size_t fragsnip_size = sizeof(sixlowpan_frag_t) +
                                 ((sixlowpan_frag_1_is((sixlowpan_frag_t *)frag))
                                 ? 0U : sizeof(frag->offset));
```